consensus/validator.go

```diff
@@ -59,11 +61,38 @@ func (consensus *Consensus) onAnnounce(msg *msg_pb.Message) {
 
 func (consensus *Consensus) prepare() {
 	groupID := []nodeconfig.GroupID{nodeconfig.NewGroupIDByShardID(nodeconfig.ShardID(consensus.ShardID))}
-	for _, key := range consensus.priKey {
+	priKeys := []*bls.PrivateKeyWrapper{}
+	for i, key := range consensus.priKey {
 		if !consensus.IsValidatorInCommittee(key.Pub.Bytes) {
 			continue
 		}
-		networkMessage, err := consensus.construct(msg_pb.MessageType_PREPARE, nil, &key)
+		priKeys = append(priKeys, &consensus.priKey[i])
+		if !consensus.MultiSig {
+			networkMessage, err := consensus.construct(msg_pb.MessageType_PREPARE, nil, []*bls.PrivateKeyWrapper{&key})
```

Comment:
can constructing the message and sending be refactored out to a function? lot of duplicate code... :(

Reply:
Originally, planned to leave it as code to delete once migrated. But may leave it as a option. So refactored.